driver-async/src/main/com/mongodb/async/client/FindIterableImpl.java

```diff
@@ -90,15 +89,11 @@
 
     @Override
     public FindIterable<TResult> batchSize(final int batchSize) {
+        super.batchSize(batchSize);
         findOptions.batchSize(batchSize);
         return this;
     }
 
-    @Override
-    public Integer getBatchSize() {
```

Comment:
No need for an override and `getBatchSize()` can now return `null`, fixing the issue with Observers in #238